packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -0,0 +1,611 @@
+
+export interface ClusterConfig {
+  readonly masterNodes: number;
+  readonly masterNodeInstanceType: string;
+  readonly dataNodes: number;
+  readonly dataNodeInstanceType: string;
+  /**
+   * The number of AZs that you want the domain to use. When you enable zone
+   * awareness, Amazon ES allocates the nodes and replica index shards that
+   * belong to a cluster across the specified number of Availability Zones (AZs)
+   * in the same region to prevent data loss and minimize downtime in the event
+   * of node or data center failure. Don't enable zone awareness if your cluster
+   * has no replica index shards or is a single-node cluster.
+   *
+   * @default - Zone awareness is not enabled.
+   */
+  readonly availabilityZoneCount?: number;
+}
+
+export interface EbsOptions {
+  readonly iops?: number;
+  readonly volumeSize: number;
+  readonly volumeType: EbsDeviceVolumeType;
+}
+
+export interface LoggingOptions {
+  /**
+   * Specify if slow search logging should be set up.
+   *
+   * @default - false
+   */
+  readonly slowSearchLogEnabed?: boolean;
+
+  /**
+   * Log slow searches to this log group.
+   *
+   * @default - a new log group is created if slow search logging is enabled
+   */
+  readonly slowSearchLogGroup?: logs.LogGroup;
+
+  /**
+   * Specify if slow index logging should be set up.
+   *
+   * @default - false
+   */
+  readonly slowIndexLogEnabed?: boolean;
+
+  /**
+   * Log slow indecies to this log group.
+   *
+   * @default - a new log group is created if slow index logging is enabled
+   */
+  readonly slowIndexLogGroup?: logs.LogGroup;
+
+  /**
+   * Specify if Elasticsearch application logging should be set up.
+   *
+   * @default - false
+   */
+  readonly appLogEnabed?: boolean;
+
+  /**
+   * Log Elasticsearch application logs to this log group.
+   *
+   * @default - a new log group is created if app logging is enabled
+   */
+  readonly appLogGroup?: logs.LogGroup;
+}
+
+/**
+ * Whether the domain should encrypt data at rest, and if so, the AWS Key
+ * Management Service (KMS) key to use. Can only be used to create a new domain,
+ * not update an existing one
+ */
+export interface EncryptionAtRestOptions {
+  /**
+   * Specify true to enable encryption at rest.
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * Supply if using KMS key for encryption at rest.
+   */
+  readonly kmsKey?: kms.Key;
+}
+
+/**
+ * Properties for a AWS Elasticsearch Domain.
+ */
+export interface DomainProps {
+  /**
+   * Domain Access policies.
+   *
+   * @default - No access policies.
+   */
+  readonly accessPolicies?: PolicyStatement[];
+
+  /**
+   * Additional options to specify for the Amazon ES domain.
+   *
+   * @default - no advanced options are specified
+   */
+  readonly advancedOptions?: { [key: string]: (string) };
+
+  /**
+   * `AWS::Elasticsearch::Domain.CognitoOptions`
+   */
+  readonly cognitoOptions?: CfnDomain.CognitoOptionsProperty | cdk.IResolvable;
```

Comment:
Don't expose `Cfn` classes on the L2 level. Its ok to duplicate and create a new interface.

Reply:
I have created a new interface for the configuration. Since Cognito doesn't have L2 identity pool and L2 user pool is preview I am referencing the pool identities as string instead of using the cognito.UserPool or cognito.CfnIdentityPool.